client/components/calypso-i18n-provider/index.tsx

```diff
@@ -0,0 +1,31 @@
+/**
+ * External dependencies
+ */
+import React, { useEffect } from 'react';
+import { setLocaleData as setWpI18nLocaleData } from '@wordpress/i18n';
+import { I18nProvider } from '@automattic/react-i18n';
+import i18n from 'i18n-calypso';
+
+const CalypsoI18nProvider: React.FunctionComponent = ( { children } ) => {
+	const [ localeData, setLocaleData ] = React.useState( i18n.state.locale );
+
+	useEffect( () => {
+		const onChange = () => {
+			// Create a new object to ensure the provider sees the data has changed
+			const nextLocaleData = Object.assign( {}, i18n.state.locale );
```

Comment:
Let's use a getter method (`i18n.getLocale()`) instead of direct access to `i18n.state.locale`.